app/models/caches/plan_details.rb

```diff
@@ -1,7 +1,13 @@
 module Caches
   class PlanDetails
-    # TODO: Implemented by 16310
     def self.lookup_rate_with_area(plan_id, rate_schedule_date, effective_age, rating_area)
+      calc_age = age_bounding(plan_id, effective_age)
+      age_record = $plan_age_lookup[plan_id][calc_age].detect do |pt|
```

Comment:
We should actually re-structure this as another global that is a separate hash with 3 keys.  You'll build it in the loading method alongside the existing lookup hash which should remain unchanged.